### ISSUE TITLE
If Letter grade type (3) previous set, stack trace on settings

### ISSUE THREAD

**Issue by jonespm** [sakaiproject/sakai#4355]
Relates to #2730. Previous version of GB allowed a letter grade type which GBNG doesn't support. From what I can tell, if this value is set in the database gb_gradebook_t->grade_type (3) then neither of the radio buttons on the settings page are selected. If the instructor changes any other option here and clicks save they are presented with a stacktrace. (Cut for brevity since it's relatively easy to reproduce by changing value in GB)

I can think of a few possible fixes. 

- Change GradebookInformation to use Integer instead of int values so that a null is applicable here. A null would probably need to map to some value though so maybe not a great solution. 
- Have a validator that won't let you save without a radio selected (even if LETTER was previously selected)
- Have a disabled radio that still shows letter as a value but is not normally selectable and if selected presents the user with a warning that they need to change it?

Maybe you have some other ideas to let through unsupported values. I don't think this is a big deal as only a few schools with old courses would encounter this, but it has come up a few times for some that are still using both GB2 and GBNG and don't want to convert this type over.

```
org.apache.wicket.WicketRuntimeException: Method onFormSubmitted of interface org.apache.wicket.markup.html.form.IFormSubmitListener targeted at [Form [Component id = form]] on component [Form [Component id = form]] threw an exception
	at org.apache.wicket.RequestListenerInterface.internalInvoke(RequestListenerInterface.java:268)
... 
Caused by: org.apache.wicket.util.convert.ConversionException: Method [public int org.sakaiproject.service.gradebook.shared.GradebookInformation.getGradeType()]. Can't convert null value to a primitive class: int for setting it on org.sakaiproject.service.gradebook.shared.GradebookInformation@aedfd65[
  selectedGradingScaleUid=PassNotPassMapping
```

**Comment by steveswinsburg** [comment]
I think if an out of range value is given then it should just default to something or at least validate that nothing was selected.

**Comment by jonespm** [comment]
Yeah, I think it would be fine if it defaulted, as long as it put up a notice to the user that their grading scheme was changed. They could change it back again, but we have a lot of schools still using GB1/GB2 because there are a few features they can't give up, like grade export and letter grading.
